caffe2/core/nomnigraph/test.cc

```diff
@@ -0,0 +1,692 @@
+      return a_->getName() == b_->getName();
+    }
+    return sameKind;
+  }
+};
+
+
+auto bbprinter = [](typename nom::repr::NNCFGraph::NodeRef node) {
+  std::map<std::string, std::string> labelMap;
+  assert(node->data() && "Node doesn't have data, can't render it");
+  auto *bb = dyn_cast<nom::repr::BasicBlockType<nom::repr::NNGraph>>(
+      node->data().get());
+  labelMap["label"] = std::to_string((unsigned long long)node) + "\\n";
+  for (const auto &instr : bb->getInstructions()) {
+    assert(isa<nom::repr::NeuralNetOperator>(instr->data()) &&
+           "Invalid instruction.");
+    auto *op = dyn_cast<nom::repr::NeuralNetOperator>(instr->data().get());
+    bool hasOutput = false;
+    for (const auto &outEdge : instr->getOutEdges()) {
+      auto *output =
+          dyn_cast<nom::repr::NeuralNetData>(outEdge->head()->data().get());
+      labelMap["label"] += " " + output->getName();
+      hasOutput = true;
+    }
+    if (hasOutput) {
+      labelMap["label"] += " = ";
+    }
+    labelMap["label"] += op->getName();
+    for (const auto &inEdge : instr->getInEdges()) {
+      auto *arg =
+          dyn_cast<nom::repr::NeuralNetData>(inEdge->tail()->data().get());
+      labelMap["label"] += " " + arg->getName();
+    }
+    labelMap["label"] += "\\l";
+  }
+  labelMap["shape"] = "box";
+  return labelMap;
+};
+
+auto cfgedgeprinter = [](typename nom::repr::NNCFGraph::EdgeRef edge) {
+  std::map<std::string, std::string> labelMap;
+  if (edge->data() == -1) {
+    labelMap["label"] = "F";
+  } else if (edge->data() == 1) {
+    labelMap["label"] = "T";
+  }
+  return labelMap;
+};
+
+auto nnprinter = [](typename nom::repr::NNGraph::NodeRef node) {
+  std::map<std::string, std::string> labelMap;
+  assert(node->data() && "Node doesn't have data, can't render it");
+  if (isa<nom::repr::NeuralNetOperator>(node->data())) {
+    auto *op = dyn_cast<nom::repr::NeuralNetOperator>(node->data().get());
+    labelMap["label"] =
+        op->getName() + " (" + std::to_string((unsigned long long)node) + ")";
+    auto *annotation = op->getAnnotation();
+    if (annotation && isa<nom::repr::DeviceAnnotation>(annotation)) {
+      auto device_annotation =
+          dyn_cast<nom::repr::DeviceAnnotation>(annotation);
+      labelMap["label"] += "\\n[" + device_annotation->getDevice() + "]";
+      auto hash = std::hash<std::string>{}(device_annotation->getDevice());
+      std::stringstream hex_stream;
+      hex_stream << std::hex << hash;
+      labelMap["color"] = "#" + hex_stream.str().substr(0, 6);
+      labelMap["fontcolor"] = labelMap["color"];
+    }
+    labelMap["shape"] = "box";
+  } else if (isa<nom::repr::Data>(node->data())) {
+    auto tensor = dyn_cast<nom::repr::NeuralNetData>(node->data().get());
+    labelMap["label"] = tensor->getName();
+    labelMap["label"] += "_" + std::to_string(tensor->getVersion()) + " " + std::to_string((unsigned long long)node);
+  }
+  return labelMap;
+};
+
+int main(int argc, char *argv[]) {
```

Comment:
Please use Google Test for the unit tests

Reply:
this whole test file is going to be formalized in a later PR (and actually migrated to caffe2 test locations in caffe2/opt)